dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -1321,6 +1321,17 @@
 		</ItemGroup>
 	</Target>
 
+	<Target Name="_ExcludeUnusedFilesFromTheBundle"
+		Condition="'$(_UseNativeAot)' == 'true'"
+		AfterTargets="_LinkNativeExecutable"
+		BeforeTargets="CopyFilesToPublishDirectory">
+		<ItemGroup>
+			<!-- We don't ship assemblies with NativeAOT -->
+			<!-- Exclude assemblies explicitly until https://github.com/dotnet/runtime/issues/87187 is not resolved -->
+			<ResolvedFileToPublish Remove="@(ResolvedFileToPublish)" Condition="'%(ResolvedFileToPublish.Extension)' == '.dll'" />
```

Comment:
This will likely be problematic on macOS where native libraries are part of the bundle (eg. libe_sqlite3.dylib from NuGet, but also possibly libSystem.Native* from runtime itself). On iOS-like platforms this may also be a problem with .framework/.xcframework coming from NuGets (eg. SkiaSharp).

Reply:
I am thinking whether there is a meta property on the ItemGroup telling whether the file came from NativeLibrary (or equivalent) originally...